src/vs/workbench/contrib/comments/browser/commentService.ts

```diff
@@ -16,9 +17,15 @@ import { ICellRange } from 'vs/workbench/contrib/notebook/common/notebookRange';
 import { IWorkbenchLayoutService } from 'vs/workbench/services/layout/browser/layoutService';
 import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
 import { COMMENTS_SECTION, ICommentsConfiguration } from 'vs/workbench/contrib/comments/common/commentsConfiguration';
+import { IContextKey, IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
 
 export const ICommentService = createDecorator<ICommentService>('commentService');
 
+export const WorkspaceHasCommenting = new RawContextKey<boolean>('workspaceHasCommenting', false, {
+	description: nls.localize('hasCommentingProvider', "Whether the open workspace has either comments or commenting ranges."),
```

Comment:
Should this be "comment provider"?

Reply:
Which part?

Reply:
Oh never mind, I see it's `CommentingRangeProvider` in the API. Did a search for commentprovider which showed results which is what my comment is based on.